receiver/bigipreceiver/client.go

```diff
@@ -0,0 +1,345 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bigipreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver"
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver/internal/models"
+)
+
+// loginPath is the path to the login endpoint
+const loginPath = "/mgmt/shared/authn/login"
+
+// virtualServersPath is the path to the virtual servers endpoint
+const virtualServersPath = "/mgmt/tm/ltm/virtual"
+
+// virtualServersStatsPath is the path to the virtual servers statistics endpoint
+const virtualServersStatsPath = "/mgmt/tm/ltm/virtual/stats"
+
+// poolsStatsPath is the path to the pools statistics endpoint
+const poolsStatsPath = "/mgmt/tm/ltm/pool/stats"
+
+// nodesStatsPath is the path to the nodes statistics endpoint
+const nodesStatsPath = "/mgmt/tm/ltm/node/stats"
+
+// poolMembersStatsPathSuffix is the suffix added onto an individual pool's statistics endpoint
+const poolMembersStatsPathSuffix = "/members/stats"
+
+// custom errors
+var (
+	errCollectedNoPoolMembers = errors.New(`All pool member requests have failed`)
```

Comment:
Nit: errors in go generally aren't capitalized

Reply:
@BinaryFissionGames changed